OpenSim/Simulation/Model/Model.h

```diff
@@ -105,6 +105,27 @@ class PhysicalOffsetFramesFormLoop : public Exception {
     }
 };
 
+class JointFramesHaveSameBaseFrame : public Exception {
+public:
+    JointFramesHaveSameBaseFrame(const std::string& file,
+        size_t line,
+        const std::string& func,
+        const std::string& thisName,
+        const std::string& parentName,
+        const std::string& childName,
+        const std::string& baseName) :
+        Exception(file, line, func) {
+        std::string msg = "Joint '" + thisName + 
+            "' cannot connect parent frame '" +
+            parentName + "' to child frame '" + childName + "'.\n"
+            "'Parent and child frames have the same base frame.'"
```

Comment:
I think there are a couple extra punctuation marks (beginning and end), but feel free to ignore (or push with `[ci skip]`):
`"'Parent and child frames have the same base frame.'"` -> `"Parent and child frames have the same base frame '"`